### ISSUE TITLE
Retweet/Reblog with comment

### ISSUE THREAD

**Issue by yvolk** [andstatus/andstatus#266]
There was a discussion at Quitter.se on this: https://quitter.se/notice/3715247
which left me with a feeling that that talking about the same subject different participants kept different usage scenarios / different things in mind.
Before anything can be done, we need to clarify meaning and implementation as it is visible to the User.
Here are points, which need to be thought of together:
1. Currently "Reblog with comment" may be done in two steps: "Reblog" + "Reply". This how usually Social networks work. Technically "Retweet /Reblog" API (communication protocol) doesn't allow addition of comments. So do we really need to automate this two-step process?
2.  On the other hand, recently Twitter added separate "Retweet with comment" feature: http://techcrunch.com/2015/04/06/retweetception/  Is _that_ actually a combination of the above two steps or is this a new operation? 
3. Anyway, GNU Social doesn't support this as a "single action" meaning that these will be _two_ separate things in e.g. GNU Social Web interface: Reblog and Reply.
4. Or do we really need "Reply with Quote"? @mcscx seems like shows us an example of a "Re-Dent / Reblog" in this thread. But when he writes: 
```
@helioloureiro you could do "Share" > "Andstatus", then modify the original post 
RD @helioloureiro no such operation here
```
the text after "RD @helioloureiro" (which should presumably mean "Reblog of the below post by @helioloureiro") is _not_ a complete copy of the original post: 
```
@mcscx no such operation here :( https://quitter.se/attachment/950194
```
So the above example is really a _Quote_ (of some part of the original post), not a "repeat".
5. And finally, a question to the author of the first message in this discussion: It's interesting that having account at the site with 140 characters limit you are asking for the feature. I don't see much practical use of this as usually there is almost no space left for the "comment". Moreover, "Reblog with comment of Reblog with comment" will certainly be impossible with low character limits...

**Comment by akater** [comment]
> I don't see much practical use of this as usually there is almost no space left for the "comment".

In case you're interested, I was using «Retweet with comment» regularly in TweetCaster, and I do miss it in AndStatus.

**Comment by moshpirit** [comment]
You can use: Share / AndStatus / Twitter

**Comment by yvolk** [comment]
@akater Do you mean that even, say, 20 characters for a comment are enough for you? Or I don't understand how this works?
Could you give concrete example?

**Comment by akater** [comment]
«Retweet with comment» was present in TweetCaster long before April 2014*, and used to look like this: [Comment] RT \@originalAuthor [Original message]

Some examples (including false positives, due to limitations of Twitter's search facility): https://twitter.com/search?q=RT%20from%3Aakater%20since%3A2014-01-01%20until%3A2015-06-30&src=typd
_______
* Apparently, it was made official in April 2014 http://mashable.com/2015/04/07/twitter-retweet-comments/

**Comment by yvolk** [comment]
I didn't find Twitter API  for the Retweet with comments. 
"Simple" Retweet: https://dev.twitter.com/rest/reference/post/statuses/retweet/%3Aid
In Twitter docs I see https://support.twitter.com/articles/20169873# , which says:  "A Tweet that you forward to your followers is known as a Retweet. You can use our built-in Retweet feature for this. If you'd like to add your own comments to the tweet, you can use the quote Tweet feature"
So maybe "Retweet with comment" is really a Retweet + a special form of a Reply.

**Comment by Yowlen** [comment]
I discovered the answer to this issue. The feature mentioned here (retweet with comment) is, in fact, different from the standard "quote tweet" thing. However, it's far simpler than you make it out to be. Basically, it's just tacking on a link to the tweet you're retweeting. Twitter's site/app takes care of pulling up the destination & displaying it, just like other links. As a result, in order to support this in AndStatus, issue #4 will need to be implemented first.

**Comment by yvolk** [comment]
@Yowlen Your example looks almost like a reply with a replied message being visually embedded into the reply. This understanding of the feature allows to use/interprete _any_ Reply as a "Retweet with comment": 
we just need to implement such visual presentation in a timeline.
?!
And this looks smarter than Twitter invented :-)

**Comment by Yowlen** [comment]
My example is the same as if I used the "Retweet with comment" feature mentioned in the techcrunch link in the original comment, except I created it by manually copy/pasting the tweet url in. It was mostly thanks to the mobile site and app that I was able to figure this out, since the quoted tweets would show in the app, but the mobile site just shows links.

As for that last comment, it's exactly what Twitter does. Twitter just worded their explanation in an extremely dumb way. ;)

That's why there's no API for it. In fact, they've expanded the capability to fetch the previews of other links, which, ironically, does work on the mobile site, unlike the initial feature it was based on. lol

**Comment by yvolk** [comment]
Thinking of this another time I see that there is a semantic difference between a simple Reply and a Retweet with a Comment: I don't think it will be correct to show every Reply with a replied tweet embedded. 
So we better create both Retweet and Reply in one operation and let AndStatus present such a pair of tweets as one combined "Retweet with Comment".

**Comment by Yowlen** [comment]
Exactly. It may not be entirely implemented on other platforms, so it's good to have the legacy option for compatibility.

For the record, Twitter has actually eliminated the original way to "Quote Tweet" in the official apps/site, except on their mobile site. Funny how they've neglected this feature entirely on the mobile site, when their mobile site is still much more polished than their mobile app. XD

**Comment by yvolk** [comment]
@Yowlen Yes, Twitter developers are ordinary people trying to do their best (I hope).
Neglection of some former decisions may mean that that decisions badly affected on the global Twitter performance, so they had to disable them...

**Comment by Yowlen** [comment]
@yvolk As I use Twitter more & more, it's increasingly clear they're like Wal-Mart & other big corporations. Profits are their top priority, not customers. Despite outrage by many Twitter users, they still implemented a lot of features that promotes specific content in users' timelines, & recent censoring even shows whose payroll they're on. The # WhichHillaryCensored hashtag, namely. As a result, the devs likely either don't care about users, or are underpaid to where they simply can't care, & end up making poor decisions.

**Comment by yaomtc** [comment]
Currently, Quote Retweets are just shown in the timelime as a t.co link, and the tweet they're quoting isn't shown. Is there any way to get around this?

**Comment by yvolk** [comment]
@yaomtc You can try "Open conversation" context menu item: maybe it shows the quoted Tweet in the Conversation view of AndStatus?!

**Comment by yaomtc** [comment]
It does not.
